rules/no-instanceof-builtin-object.js

```diff
@@ -0,0 +1,174 @@
+
+	return {
+		/** @param {import('estree').BinaryExpression} node */
+		'BinaryExpression[operator="instanceof"]'(node) {
+			if (node.right.type !== 'Identifier') {
+				return;
+			}
+
+			const {left, right} = node;
+
+			let tokenStore = sourceCode;
+			let instanceofToken = tokenStore.getTokenAfter(left, isInstanceofToken);
+			if (!instanceofToken && sourceCode.parserServices.getTemplateBodyTokenStore) {
+				tokenStore = sourceCode.parserServices.getTemplateBodyTokenStore();
+				instanceofToken = tokenStore.getTokenAfter(left, isInstanceofToken);
+			}
+
+			if (primitiveConstructors.has(right.name) || right.name === 'Function') {
+				// Check if the node is in a Vue template expression
+				const vueExpressionContainer = sourceCode.getAncestors(node).findLast(ancestor => ancestor.type === 'VExpressionContainer');
+
+				// Get safe quote
+				const safeQuote = vueExpressionContainer ? (sourceCode.getText(vueExpressionContainer)[0] === '"' ? '\'' : '"') : '\'';
+
+				context.report({
+					node,
+					messageId: MESSAGE_ID,
+					* fix(fixer) {
+						yield * fixSpaceAroundKeyword(fixer, node, sourceCode);
+
+						const leftRange = getParenthesizedRange(left, tokenStore);
+						yield fixer.insertTextBeforeRange(leftRange, 'typeof ');
+
+						yield fixer.replaceText(instanceofToken, '===');
+
+						const rightRange = getParenthesizedRange(right, tokenStore);
+
+						yield fixer.replaceTextRange(rightRange, safeQuote + sourceCode.getText(right).toLowerCase() + safeQuote);
+					},
+				});
+			} else if (referenceConstructors.has(right.name)) {
+				if (right.name === 'Array') {
+					context.report({
+						node,
+						messageId: MESSAGE_ID,
+						* fix(fixer) {
+							yield * fixSpaceAroundKeyword(fixer, node, sourceCode);
+
+							const range = getParenthesizedRange(left, tokenStore);
+							yield fixer.insertTextBeforeRange(range, 'Array.isArray(');
+							yield fixer.insertTextAfterRange(range, ')');
+
+							yield * replaceNodeOrTokenAndSpacesBefore(instanceofToken, '', fixer, sourceCode, tokenStore);
+							yield * replaceNodeOrTokenAndSpacesBefore(right, '', fixer, sourceCode, tokenStore);
+						},
+					});
+				} else if (right.name === 'Error' && shippedProposals) {
+					context.report({
+						node,
+						messageId: MESSAGE_ID,
+						* fix(fixer) {
+							yield * fixSpaceAroundKeyword(fixer, node, sourceCode);
+
+							const range = getParenthesizedRange(left, tokenStore);
+							yield fixer.insertTextBeforeRange(range, 'Error.isError(');
+							yield fixer.insertTextAfterRange(range, ')');
+
+							yield * replaceNodeOrTokenAndSpacesBefore(instanceofToken, '', fixer, sourceCode, tokenStore);
+							yield * replaceNodeOrTokenAndSpacesBefore(right, '', fixer, sourceCode, tokenStore);
+						},
+					});
+				} else {
+					context.report({node, messageId: MESSAGE_ID});
+				}
+			}
+		},
+	};
+};
+
+const schema = [
+	{
+		type: 'object',
+		properties: {
+			shippedProposals: {
```

Comment:
Something like `useErrorIsError: true` is easier to understand.

Reply:
This is based on [Babel's parameter](https://babeljs.io/docs/babel-preset-env#shippedproposals) design

If others believe this parameter is unsuitable, we can consider using an alternative instead.

Reply:
> once Error.isError is officially incorporated into ECMAScript, the option useErrorIsError: true will no longer be necessary.

We did this a lot.